File: backend/bilhetesja-api/bilhetesja-api/Entities/Event.cs
using System.ComponentModel.DataAnnotations;

namespace bilhetesja_api.Entities
{
    public enum StatusEvento
    {
        Pendente,
        Aprovado,
        Rejeitado
    }

    public class Event
    {
        public int Id { get; set; }

        [Required, MaxLength(255)]
        public required string Nome { get; set; }

        public string Descricao { get; set; } = string.Empty;

        [MaxLength(255)]
        public string Local { get; set; } = string.Empty;

        public DateTime DataEvento { get; set; }

        public int LotacaoTotal { get; set; } = 0;

        public StatusEvento Status { get; set; } = StatusEvento.Aprovado;
        public int CategoriaId { get; set; }
        public Category? Categoria { get; set; }

        public int OrganizadorId { get; set; }
        public User? Organizador { get; set; }

        public int? ImagemId { get; set; }
        public Image? Imagem { get; set; }

        public ICollection<TicketType>? TiposBilhetes { get; set; } = new List<TicketType>();
    }

}

File: web/src/domain/models/event.ts

export type Event = {
  id: number;
  nome: string;
  categoriaId: number;
  descricao: string;
  dataEvento: string;
  lotacaoTotal: number;
  status: number;
  local: string;
  nomeCategoria?: string;
  categoria: {
    id: number;
    nome: string;
  };
  organizador: {
    id: number;
    nome: string;
    email: string;
  };
  imagem?: string | null; // Optional field for image URL
  tiposBilhetes?: ticket[];
}



export type ticket = {
  id: number;
  nome: string;
  preco: number;
  quantidade: number;
  eventoNome: string;
  eventoId: number;
  event: Event
}
